### PR TITLE
scripts: move API from core

### PR DESCRIPTION
**Pull request by thc202**
Move the API from core, which can already be provided by the add-on.

Part of zaproxy/zaproxy#8027.

### CODE REVIEW

**Review by kingthorin — APPROVED** [review]

File: addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/ScriptApi.java [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+                    params.getString(PARAM_VAR_KEY), params.getString(PARAM_VAR_VALUE));
+            return ApiResponseElement.OK;
+        } else if (ACTION_SET_SCRIPT_VAR.equals(name)) {
+            ScriptVars.setScriptVar(
+                    getAndValidateScriptName(params),
+                    params.getString(PARAM_VAR_KEY),
+                    params.getString(PARAM_VAR_VALUE));
+            return ApiResponseElement.OK;
+        } else {
+            throw new ApiException(ApiException.Type.BAD_ACTION);
```
Comment (thc202) [inline addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/ScriptApi.java:439]: This line was changed compared to core code, previously `BAD_VIEW`.

Comment (kingthorin) [reply addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/ScriptApi.java:439]: 😄 good catch

Comment (thc202) [reply addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/ScriptApi.java:439]: Tests caught it :)

**Comment by thc202** [comment]
Updated to use the correct method when checking if the core API is deprecated.

**Review by kingthorin — APPROVED** [review]

**Review by ricekot — APPROVED** [review]
Thank you!

**Pull request merged**